### PR TITLE
SpanData::$parent property

### PR DESCRIPTION
**Pull request by krakjoe**
Implement access to SpanData::$parent.

  - This does not yet use ZEND_ACC_READONLY to keep the source simple (php9 will use ZEND_ACC_READONLY)
  - This does not prohibit indirect modification, that would be costly to all other reads and doesn't seem worth it

### CODE REVIEW

File: ext/php7/ddtrace.c [hunk 1]
```diff
@@ -282,6 +282,28 @@ static zend_object *ddtrace_span_data_clone_obj(zval *old_zv) {
     return new_obj;
 }
 
+#if PHP_VERSION_ID >= 70400
+static zval* ddtrace_span_data_readonly(zval *object, zval *member, zval *value, void **cache_slot) {
+#else
+static void ddtrace_span_data_readonly(zval *object, zval *member, zval *value, void **cache_slot) {
+#endif
+    if (Z_TYPE_P(member) == IS_STRING && zend_string_equals_literal(Z_STR_P(member), "parent")) {
+        zend_throw_exception_ex(zend_ce_exception, 0,
```
Comment (bwoebi) [inline ext/php7/ddtrace.c:291]: Is there a reason why we use exception here? For PHP 5, sure, but in general Error is appropriate I'd say?

File: tests/ext/span_with_parent_property_access.phpt [hunk 2]
```diff
@@ -15,10 +15,6 @@ if (!isset($root->parent) && $span->parent === $root) {
         $span->parent = null;
     } catch (\Exception $error) {
         print $error->getMessage() . PHP_EOL;
-    } finally {
-        if ($span->parent === null) {
```
Comment (bwoebi) [inline tests/ext/span_with_parent_property_access.phpt]: btw. you can keep that if, given that the catch block doesn't rethrow anyway, just dropping finally {}

File: ext/php7/ddtrace.c (hunk 1 above)
Comment (krakjoe) [reply ext/php7/ddtrace.c:291]: It started out Error, but the test is for all versions, and the only compatible test catches Exception.

File: tests/ext/span_with_parent_property_access.phpt (hunk 2 above)
Comment (krakjoe) [reply tests/ext/span_with_parent_property_access.phpt]: 5.5 is first version with finally {}

File: ext/php7/ddtrace.c (hunk 1 above)
Comment (bwoebi) [reply ext/php7/ddtrace.c:291]: you could do `} catch (\Exception) { print $error->getMessage() . PHP_EOL; } catch (\Throwable $e) { print $error->getMessage() . PHP_EOL; }` in the test

Comment (krakjoe) [reply ext/php7/ddtrace.c]: I'd like to have a conversation about this.

**Review by bwoebi — APPROVED** [review]
Looks good to me now, thanks Joe :-)

**Pull request merged**